tests/test_helper/bats/upgrade_dp.bats

```diff
@@ -0,0 +1,54 @@
+# This is a bash shell fragment -*- bash -*-
+
+# The test simulates instance(s) by creating Linux namespaces inside the test
+# containers.
+export TEST_LXD_LAUNCH_ARGS="-c security.nesting=true"
+
+load "test_helper/setup_teardown/$(basename "${BATS_TEST_FILENAME//.bats/.bash}")"
+
+setup() {
+    load test_helper/lxd.bash
+    load test_helper/common.bash
+    load test_helper/microovn.bash
+    load test_helper/upgrade_procedures.bash
+    load ../.bats/bats-support/load.bash
+    load ../.bats/bats-assert/load.bash
+
+    # Ensure TEST_CONTAINERS is populated, otherwise the tests below will
+    # provide false positive results.
+    assert [ -n "$TEST_CONTAINERS" ]
+
+    # Set up namespace for data path test
+    for container in $CHASSIS_CONTAINERS; do
+        microovn_add_gw_router "$container"
+        netns_add "$container" upgrade_dp0
+        microovn_add_vif "$container" upgrade_dp0 upgrade_vif0
+    done
+}
+
+teardown() {
+    # Remove namespace used for data path test
+    for container in $CHASSIS_CONTAINERS; do
+        microovn_delete_vif "$container" upgrade_dp0 upgrade_vif0
+        netns_delete "$container" upgrade_dp0
+        microovn_delete_gw_router "$container"
+    done
+}
+
+
+@test "MicroOVN control of schema conversion prevents data path downtime" {
```

Comment:
Just a general thought here. We probably want to run this DP test on any upgrade, not just those that involve OVN schema upgrades. I'd say it's even more critical to ensure that in-channel snap upgrades don't cause DP downtime.
Otherwise this LGTM so far.

Reply:
Thanks for feedback.

On in-channel upgrade tests: That would be nice, the upgrade tests are currently hard wired for cross channel checks though:
https://github.com/canonical/microovn/blob/833634c6a724d177a01ddef2a2957427ef26ce63/tests/test_helper/setup_teardown/upgrade.bash#L13